### PR TITLE
Avoid indirect calls and enforce strict & warnings

### PR DESCRIPTION
**Pull request by atoomic**
This work was performed on the p7 experimental branch
and most of the changes comes from kid51.

### CODE REVIEW

File: parts/inc/ppphtest [hunk 1]
```diff
@@ -680,7 +680,7 @@ my %p;
 my $fail = 0;
 for (@o) {
   my($name, $flags) = /^(\w+)(?:\s+\[(\w+(?:,\s+\w+)*)\])?$/ or $fail++;
-  exists $p{$name} and $fail++;
+  { no warnings 'uninitialized'; exists $p{$name} and $fail++; }
```
Comment (khwilliamson) [inline parts/inc/ppphtest:683]: This module is the only one that has to work in really early perls.  That means, before you could say 'no warnings'.  The paradigm elsewhere in the code appears to be
BEGIN { 'warnings'->unimport('uninitialized') if "$]" > '5.006' }

**Comment by jkeenan** [comment]
> This is failing tests on early perls due to the warnings issue that I commented on

Would something like this work (modulo line breaks)?
```
diff --git a/dist/Devel-PPPort/parts/inc/ppphtest b/dist/Devel-PPPort/parts/inc/ppphtest
index c6877afcbe..4a8b617cb2 100644
--- a/dist/Devel-PPPort/parts/inc/ppphtest
+++ b/dist/Devel-PPPort/parts/inc/ppphtest
@@ -681,7 +681,7 @@ my $fail = 0;
 for (@o) {
   my($name, $flags) = /^(\w+)(?:\s+\[(\w+(?:,\s+\w+)*)\])?$/ or $fail++;
   {
-    no warnings 'uninitialized';
+    BEGIN { 'warnings'->unimport('uninitialized') if "$]" > '5.006' }
     exists $p{$name} and $fail++;
     $p{$name} = defined $flags ? { map { ($_ => 1) } $flags =~ /(\w+)/g } : '';
   }
```

**Comment by khwilliamson** [comment]
On 7/19/20 12:09 PM, James E Keenan wrote:
>     This is failing tests on early perls due to the warnings issue that
>     I commented on
> 
> Would something like this work (modulo line breaks)?
> 
> |diff --git a/dist/Devel-PPPort/parts/inc/ppphtest 
> b/dist/Devel-PPPort/parts/inc/ppphtest index c6877afcbe..4a8b617cb2 
> 100644 --- a/dist/Devel-PPPort/parts/inc/ppphtest +++ 
> b/dist/Devel-PPPort/parts/inc/ppphtest @@ -681,7 +681,7 @@ my $fail = 0; 
> for (@o) { my($name, $flags) = /^(\w+)(?:\s+\[(\w+(?:,\s+\w+)*)\])?$/ or 
> $fail++; { - no warnings 'uninitialized'; + BEGIN { 
> 'warnings'->unimport('uninitialized') if "$]" > '5.006' } exists 
> $p{$name} and $fail++; $p{$name} = defined $flags ? { map { ($_ => 1) } 
> $flags =~ /(\w+)/g } : ''; } |
> 
I need to regenerate D:P anyway, so I'll see what works:

**Comment by khwilliamson** [comment]
I tested this line successfully

'warnings'->unimport('uninitialized') if "$]" > '5.006';

File: parts/inc/ppphtest (hunk 1 above)
Comment (atoomic) [reply parts/inc/ppphtest:683]: I think `no  if "$]" > '5.006, warnings 'uninitialized';` would work

**Comment by atoomic** [comment]
@khwilliamson with the last state at 68385410 this should fix 5.6

**Comment by atoomic** [comment]
@khwilliamson as discussed I've removed the extra unimport warnings 
`BEGIN { 'warnings'->unimport('uninitialized') if "$]" > '5.006' }`
52a41ab8 ready for another review

**Pull request merged**